File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/Model/Universe.java
package edu.gatech.cs2340.spacetraderredux.Model;

import android.graphics.Point;
import android.util.Log;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Random;
import java.util.Set;

public class Universe {
    private final List<SolarSystem> solarSystemList;

    Universe(){
        ArrayList<SolarSystem> solarSystemList = new ArrayList<>();
        Set<Point> usedPoints = new HashSet<>();
        Random seed = new Random();
        for (SolarSystemName name : SolarSystemName.values()) {
            Point newLocation;
            do {
                newLocation = new Point(seed.nextInt(150), seed.nextInt(100));
            } while(usedPoints.contains(newLocation));
            usedPoints.add(newLocation);

            solarSystemList.add(new SolarSystem(name, newLocation, seed));
        }

        this.solarSystemList = Collections.unmodifiableList(solarSystemList);
    }

    public List<SolarSystem> getSolarSystems() {
        return solarSystemList;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Universe containing solar systems: {");
        for (int i = 0; i < solarSystemList.size() - 1; i++) {
            sb.append(solarSystemList.get(i) + ", ");
        }
        sb.append(solarSystemList.get(solarSystemList.size() - 1) + "}");
        return sb.toString();
    }

    public void dumpToLog() {
        String logLeft = toString();
        while (logLeft.length() > 4000) {
            Log.d("INFO", logLeft.substring(0, 4000));
            logLeft = logLeft.substring(4000);
        }
        Log.d("INFO", logLeft);
    }
}

File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/domain/Universe.kt
package edu.gatech.cs2340.spacetraderredux.domain


import edu.gatech.cs2340.spacetraderredux.domain.entities.enums.SolarSystemName
import java.util.*
import javax.inject.Inject

class Universe @Inject constructor() {
    val solarSystems: List<SolarSystem>
    val totalSystems = 25

    init {
        //TODO Make first planet a good one near the center
        //TODO fuzzy spread
        //TODO dagger this
        val solarSystemList = ArrayList<SolarSystem>()
        val usedPoints = HashSet<Point>()
        val rand = Random()
        val solarSystemNamesLen = SolarSystemName.values().size
        val selectedNames = HashSet<String>()
        var counter = 0

        while(counter < totalSystems) {
            val randomIndex = rand.nextInt(solarSystemNamesLen)
            val currSolarSystem = SolarSystemName.values()[randomIndex]
            if (!selectedNames.contains(currSolarSystem.name)) {
                var newLocation: Point
                do {
                    newLocation = Point(rand.nextInt(150), rand.nextInt(100))
                } while (usedPoints.contains(newLocation))
                usedPoints.add(newLocation)
                solarSystemList.add(SolarSystem(currSolarSystem, newLocation, rand))
                counter++
            }
            selectedNames.add(currSolarSystem.name)
        }
        this.solarSystems = Collections.unmodifiableList(solarSystemList)
    }

    override fun toString(): String {
        val sb = StringBuilder()
        sb.append("Universe containing solar systems: {")
        for (i in 0 until solarSystems.size - 1) {
            sb.append(solarSystems[i].toString() + ", ")
        }
        sb.append(solarSystems[solarSystems.size - 1].toString() + "}")
        return sb.toString()
    }

    /*fun dumpToLog() {
        var logLeft = toString()
        while (logLeft.length > 4000) {
            Log.d("INFO", logLeft.substring(0, 4000))
            logLeft = logLeft.substring(4000)
        }
        Log.d("INFO", logLeft)
    }*/
}
